src/oauth2_priv_set.erl

```diff
@@ -49,40 +49,52 @@ new(Path) when is_binary(Path) ->
 %% @doc Returns the union of Set1 and Set2, i.e., a set such that
 %%      any path present in either Set1 or Set2 is also present in the result.
 -spec union(priv_set(), priv_set()) -> priv_set().
-union([H1={node, Name1, _}|T1], [H2={node, Name2, _}|T2]) when Name1 < Name2 ->
-    [H1|union(T1, [H2|T2])];
-union([H1={node, Name1, _}|T1], [H2={node, Name2, _}|T2]) when Name1 > Name2 ->
-    [H2|union([H1|T1], T2)];
-union([{node, Name, S1}|T1], [{node, Name, S2}|T2]) ->
-    [{node, Name, union(S1, S2)}|union(T1, T2)];
-union(['*'|_], _) -> ['*']; %% '*' in union with anything is still '*'.
-union(_, ['*'|_]) -> ['*'];
-union([], Set)    -> Set;
-union(Set, [])    -> Set.
+union([H1 = {node, Name1, _} | T1], [H2 = {node, Name2, _} | T2]) when Name1 < Name2 ->
+    [H1 | union(T1, [H2 | T2])];
+union([H1 = {node, Name1, _} | T1], [H2 = {node, Name2, _} | T2]) when Name1 > Name2 ->
+    [H2 | union([H1 | T1], T2)];
+union([{node, Name, S1} | T1], [{node, Name, S2} | T2]) ->
+    [{node, Name, union(S1, S2)} | union(T1, T2)];
+%% '*' in union with anything is still '*'.
+union(['*' | _], _) ->
+    ['*'];
+union(_, ['*' | _]) ->
+    ['*'];
+union([], Set) ->
+    Set;
+union(Set, []) ->
+    Set.
 
 %% @doc Return true if Set1 is a subset of Set2, i.e., if
 %%      every privilege held by Set1 is also held by Set2.
 -spec is_subset(priv_set(), priv_set()) -> boolean().
-is_subset([{node, N1, _}|_], [{node, N2, _}|_]) when N1 < N2 ->
-    false; %% This tree isn't present in Set2 as per the invariant.
-is_subset(Set1 = [{node, N1, _}|_], [{node, N2, _}|T2]) when N1 > N2 ->
+is_subset([{node, N1, _} | _], [{node, N2, _} | _]) when N1 < N2 ->
+    %% This tree isn't present in Set2 as per the invariant.
+    false;
+is_subset(Set1 = [{node, N1, _} | _], [{node, N2, _} | T2]) when N1 > N2 ->
     is_subset(Set1, T2);
-is_subset([{node, Name, S1}|T1], [{node, Name, S2}|T2]) ->
+is_subset([{node, Name, S1} | T1], [{node, Name, S2} | T2]) ->
     case is_subset(S1, S2) of
-        true  -> is_subset(T1, T2);
+        true -> is_subset(T1, T2);
         false -> false
     end;
-is_subset(['*'|_], ['*'|_]) -> true; %% '*' is only a subset of '*'.
-is_subset(_, ['*'|_])       -> true; %% Everything is a subset of '*'.
-is_subset([], _)            -> true; %% The empty set is a subset of every set.
-is_subset(_, _)             -> false.
```

Comment:
I do prefer these comments, but I guess it would be some kind of exception to a rule somewhere that would be tricky to only affect these type of comments.

Reply:
The formatter has no rules (I don't consider _max. length per line_ to be one...). You can tell it to avoid formatting (`%% erlfmt:ignore-begin`), otherwise it's opinionated.